realm/realm-library/src/main/cpp/java_binding_context.cpp

```diff
@@ -17,18 +17,40 @@
 #include "java_binding_context.hpp"
 #include "jni_util/java_method.hpp"
 
+#include "util.hpp"
+
 using namespace realm;
 using namespace realm::_impl;
 using namespace realm::jni_util;
 
-void JavaBindingContext::changes_available()
+void JavaBindingContext::before_notify()
 {
+    if (JniUtils::get_env()->ExceptionCheck()) {
+        return;
```

Comment:
No, when it gets here, if the exception happens before, then it should definitely be thrown. Nothing will work as expected if there is an pending java exception.
`before_notify()` is called from ALooper callback, the routine is something like:

```
alloper.poolMessage(); //From java
notify(); // from JNI, ALooper callback
before_notify(); // 1
did_change(); // 2
collection callbacks; // 3
```

from above 1/2/3, if any java exception happens, we should return to the java land immediately.